crates/host/src/wasmbus/mod.rs

```diff
@@ -2481,14 +2480,20 @@ impl Host {
                 self.start_actor(entry, actor, actor_ref, count, host_id, annotations)
                     .await?;
             }
-            (hash_map::Entry::Occupied(entry), None) => {
-                self.stop_actor(entry, host_id).await?;
-            }
-            (hash_map::Entry::Occupied(entry), Some(count)) => {
+            (hash_map::Entry::Occupied(entry), _) => {
                 let actor = entry.get();
-                let mut instances = actor.instances.write().await;
+                let mut actor_instances = actor.instances.write().await;
                 let count = usize::from(count);
-                let current = instances.values().map(Vec::len).sum();
+                let current = actor_instances
+                    .iter()
+                    .filter_map(|(instance_annotations, instances)| {
+                        if annotations_match_filter(instance_annotations, &annotations) {
+                            Some(instances.len())
+                        } else {
+                            None
+                        }
+                    })
+                    .sum();
```

Comment:
Suggestion: instead of filter_map + summing the count here, store a `current_instances` that uses the `annotations_match_filter`, and a separate `current_count`. Then in the scale-down case below, we can iterate over this filtered collection (which _likely_ only has a single member) instead of all `actor_instances` again. That also removes the need for the comment and check you added below